test/extended/util/cli.go

```diff
@@ -0,0 +1,221 @@
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/admin/policy"
+	"github.com/openshift/origin/pkg/cmd/cli"
+	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
+	testutil "github.com/openshift/origin/test/util"
+	"github.com/spf13/cobra"
+)
+
+// CLI provides function to call the OpenShift CLI and Kubernetes and OpenShift
+// REST clients.
+type CLI struct {
+	verb            string
+	adminConfigPath string
+	globalArgs      []string
+	commandArgs     []string
+	finalArgs       []string
+	stdout          io.Writer
+	verbose         bool
+	cmd             *cobra.Command
+	kubeFramework   *Framework
+}
+
+// NewCLI initialize the Kubernetes E2E framework, project role bindings.
+// If runInGinko argument is set to true, this function will install BeforeEach
+// hook to Gingo test case that grant project role bindings to a namespace created
+// by the E2E test. This option must be set inside Describe() or Context()
+func NewCLI(project, configPath string, runInGinko bool) *CLI {
+	client := &CLI{}
+	client.kubeFramework = NewFramework(project)
+	if len(configPath) == 0 {
+		FatalErr(fmt.Errorf("The configPath can't be empty"))
+	}
+	client.adminConfigPath = configPath
+	if runInGinko {
+		func() { BeforeEach(client.beforeEach) }()
+	} else {
+		client.SetNamespace(project)
+	}
+	return client
+}
+
+// OsFramework returns OpenShift framework
+func (c *CLI) OsFramework() *OsFramework {
+	return NewOsFramework(c.kubeFramework.Namespace, c.AdminRESTClient())
+}
+
+// KubeFramework returns Kubernetes framework
+func (c *CLI) KubeFramework() *Framework {
+	return c.kubeFramework
+}
+
+// SetNamespace overrides the namespace set by framework
+func (c *CLI) SetNamespace(ns string) {
+	if c.kubeFramework == nil {
+		FatalErr(fmt.Errorf("The E2E framework must be initialized"))
+	}
+	c.kubeFramework.Namespace = &kapi.Namespace{ObjectMeta: kapi.ObjectMeta{Name: ns}}
+}
+
+func (c *CLI) beforeEach() {
+	if len(c.Namespace()) > 0 {
+		Logf("Adding project role bindings to %q namespace", c.Namespace())
+		c.SetupRoleBindings()
+	} else {
+		Failf("Framework does not have the namespace set")
+	}
+}
+
+// SetupRoleBindings setups a project role binding for the current namespace
+func (c *CLI) SetupRoleBindings() *CLI {
+	for _, binding := range bootstrappolicy.GetBootstrapServiceAccountProjectRoleBindings(c.Namespace()) {
+		addRole := &policy.RoleModificationOptions{
+			RoleName:            binding.RoleRef.Name,
+			RoleNamespace:       binding.RoleRef.Namespace,
+			RoleBindingAccessor: policy.NewLocalRoleBindingAccessor(c.Namespace(), c.AdminRESTClient()),
+			Users:               binding.Users.List(),
+			Groups:              binding.Groups.List(),
+		}
+		if err := addRole.AddRole(); err != nil {
+			Failf("Unable to add role binding %+v to namespace %q: %v", addRole, c.Namespace(), err)
+		}
+	}
+	return c
+}
+
+// Verbose turns on verbose messages for current command
+func (c *CLI) Verbose() *CLI {
```

Comment:
I really like this pattern of chaining methods (the same is used in the resource builder). :+1: